File: frontend/recetario/lib/data/models/categoria.dart
class Categoria {
  final String id;
  final String nombre;
  final String? descripcion;
  final String? icono;
  final int orden;
  final bool activo;
  final DateTime? createdAt;

  Categoria({
    required this.id,
    required this.nombre,
    this.descripcion,
    this.icono,
    required this.orden,
    required this.activo,
    this.createdAt,
  });

  factory Categoria.fromJson(Map<String, dynamic> json) {
    return Categoria(
      id: json['id'] as String,
      nombre: json['nombre'] as String,
      descripcion: json['descripcion'] as String?,
      icono: json['icono'] as String?,
      orden: json['orden'] as int? ?? 0,
      activo: json['activo'] as bool? ?? true,
      createdAt: json['created_at'] != null 
          ? DateTime.parse(json['created_at'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'descripcion': descripcion,
      'icono': icono,
      'orden': orden,
      'activo': activo,
      if (createdAt != null) 'created_at': createdAt!.toIso8601String(),
    };
  }

  @override
  String toString() => 'Categoria(id: $id, nombre: $nombre, orden: $orden)';
}

File: backend/internal/models/categoria.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// Categoria representa una categoría de portafolio
type Categoria struct {
	ID          uuid.UUID `json:"id" db:"id"`
	Nombre      string    `json:"nombre" db:"nombre"`
	Descripcion string    `json:"descripcion" db:"descripcion"`
	Icono       string    `json:"icono" db:"icono"`
	Orden       int       `json:"orden" db:"orden"`
	Activo      bool      `json:"activo" db:"activo"`
	CreatedAt   time.Time `json:"created_at,omitempty" db:"created_at"`
}

// CrearCategoriaRequest representa los datos para crear una categoría
type CrearCategoriaRequest struct {
	Nombre      string `json:"nombre" validate:"required"`
	Descripcion string `json:"descripcion"`
	Icono       string `json:"icono"`
	Orden       int    `json:"orden"`
}

// ActualizarCategoriaRequest representa los datos para actualizar una categoría
type ActualizarCategoriaRequest struct {
	Nombre      *string `json:"nombre,omitempty"`
	Descripcion *string `json:"descripcion,omitempty"`
	Icono       *string `json:"icono,omitempty"`
	Orden       *int    `json:"orden,omitempty"`
	Activo      *bool   `json:"activo,omitempty"`
}
